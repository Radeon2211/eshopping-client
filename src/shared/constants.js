import * as Yup from 'yup';
import { getCountries } from 'country-fns';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productConditions = {
  NEW: 'new',
  USED: 'used',
  NOT_APPLICABLE: 'not_applicable',
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([productConditions.NEW, productConditions.USED, productConditions.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
};

export const listOfAreaCodes = getCountries().map(({ name, dial }) => {
  const finalValue = `+${dial} ${name.split('(')[0].trim()}`;
  return {
    value: dial,
    label: finalValue,
  };
});

export const listOfCountries = getCountries().map(({ name }) => {
  const finalValue = name.split('(')[0].trim();
  return {
    value: finalValue,
    label: finalValue,
  };
});

export const productPages = {
  ALL_PRODUCTS: 'ALL_PRODUCTS',
  MY_PRODUCTS: 'MY_PRODUCTS',
  USER_PRODUCTS: 'USER_PRODUCTS',
};

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const orderTypes = {
  PLACED_ORDERS: 'PLACED_ORDERS',
  SELL_HISTORY: 'SELL_HISTORY',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const modalTypes = {
  ABOUT_WEBSITE: 'ABOUT_WEBSITE',
  ADD_ADMIN: 'ADD_ADMIN',
  ADD_PRODUCT: 'ADD_PRODUCT',
  BUY_PRODUCTS: 'BUY_PRODUCTS',
  CART_ITEM_ADDED: 'CART_ITEM_ADDED',
  CHANGE_NAME: 'CHANGE_NAME',
  CHANGE_EMAIL: 'CHANGE_EMAIL',
  CHANGE_USERNAME: 'CHANGE_USERNAME',
  CHANGE_PHONE_NUMBER: 'CHANGE_PHONE_NUMBER',
  CHANGE_ADDRESS: 'CHANGE_ADDRESS',
  CHANGE_PASSWORD: 'CHANGE_PASSWORD',
  CHANGE_CONTACTS: 'CHANGE_CONTACTS',
  CHANGE_DELIVERY_ADDRESS: 'CHANGE_DELIVERY_ADDRESS',
  CLEAR_CART: 'CLEAR_CART',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  DELETE_PRODUCT: 'DELETE_PRODUCT',
  EDIT_PRODUCT: 'EDIT_PRODUCT',
  LOGIN: 'LOGIN',
  REMOVE_ADMIN: 'REMOVE_ADMIN',
  RESET_PASSWORD: 'RESET_PASSWORD',
  PENDING_USER_INFO: 'PENDING_USER_INFO',
  SEND_VERIFICATION_LINK: 'SEND_VERIFICATION_LINK',
  SIGNUP: 'SIGNUP',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: `Price - ascending` },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: `Price - ascending` },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const updateCartActions = {
  INCREMENT: 'INCREMENT',
  DECREMENT: 'DECREMENT',
  NUMBER: 'NUMBER',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};

export const userStatuses = {
  ACTIVE: 'active',
  PENDING: 'pending',
};
