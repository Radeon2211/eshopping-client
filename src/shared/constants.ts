import * as Yup from 'yup';
import countryCodes from 'country-codes-list';
import { ProductCondition, ProductPhotoFieldValue } from './types/enums';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([ProductCondition.NEW, ProductCondition.USED, ProductCondition.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
  photo: Yup.mixed()
    .test('is-error', (value) => value !== ProductPhotoFieldValue.ERROR)
    .nullable(),
};

export const listOfAreaCodes = countryCodes.customArray({
  value: '{countryCallingCode}',
  label: '+{countryCallingCode} {countryNameEn}',
});

export const listOfCountries = countryCodes.customArray({
  value: '{countryNameEn}',
  label: '{countryNameEn}',
});

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: 'Price - ascending' },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: 'Price - ascending' },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig: ScrollToOptions = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};
