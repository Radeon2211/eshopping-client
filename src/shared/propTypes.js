import PropTypes from 'prop-types';

export const deliveryAddressStrings = {
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  phonePrefix: PropTypes.string,
  phoneNumber: PropTypes.string,
  street: PropTypes.string,
  zipCode: PropTypes.string,
  city: PropTypes.string,
  country: PropTypes.string,
};

export const userProfile = {
  _id: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
  contacts: PropTypes.shape({
    email: PropTypes.bool.isRequired,
    phone: PropTypes.bool.isRequired,
  }).isRequired,
  ...deliveryAddressStrings,
};

export const fullProductItem = {
  _id: PropTypes.string.isRequired,
  quantitySold: PropTypes.number.isRequired,
  buyerQuantity: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  quantity: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  condition: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
  updatedAt: PropTypes.string.isRequired,
  __v: PropTypes.number.isRequired,
  photo: PropTypes.bool.isRequired,
  seller: PropTypes.shape({
    username: PropTypes.string.isRequired,
  }).isRequired,
};

export const cartItem = {
  _id: PropTypes.string.isRequired,
  quantity: PropTypes.number.isRequired,
  product: PropTypes.shape(fullProductItem).isRequired,
};

export const orderItem = {
  _id: PropTypes.string.isRequired,
  products: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      quantity: PropTypes.number.isRequired,
      photo: PropTypes.bool.isRequired,
    }),
  ).isRequired,
  seller: PropTypes.shape({
    username: PropTypes.string.isRequired,
    phone: PropTypes.string,
    email: PropTypes.string,
  }),
  buyer: PropTypes.shape({
    username: PropTypes.string.isRequired,
  }),
  deliveryAddress: PropTypes.shape(deliveryAddressStrings).isRequired,
  overallPrice: PropTypes.number.isRequired,
  createdAt: PropTypes.string.isRequired,
  updatedAt: PropTypes.string.isRequired,
  __v: PropTypes.number.isRequired,
};

export const signupErrors = {
  email: PropTypes.string,
  username: PropTypes.string,
  password: PropTypes.string,
  ...deliveryAddressStrings,
};

export const signupTouched = {
  email: PropTypes.bool,
  username: PropTypes.bool,
  password: PropTypes.bool,
  firstName: PropTypes.bool,
  lastName: PropTypes.bool,
  phonePrefix: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
  phoneNumber: PropTypes.bool,
  street: PropTypes.bool,
  zipCode: PropTypes.bool,
  city: PropTypes.bool,
  country: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
  'react-select-3-input': PropTypes.bool,
  'react-select-4-input': PropTypes.bool,
};

export const signupValues = {
  email: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  phonePrefix: PropTypes.shape({
    value: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
  }),
  phoneNumber: PropTypes.string.isRequired,
  street: PropTypes.string.isRequired,
  zipCode: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  country: PropTypes.shape({
    value: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
  }),
};
