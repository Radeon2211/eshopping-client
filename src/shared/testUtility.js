// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const historyPageNum = (pageNumber) => ({
  listen: jest.fn(),
  location: { search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push: jest.fn(),
});

export const propsPagination = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  maxQuantityPerPage: 2,
});
