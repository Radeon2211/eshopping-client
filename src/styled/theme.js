const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#666',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#00A790',
    greenDark: '#009680',
    red: '#F22613',
  },
  shadows: {
    level1: '0 0 2px rgba(0, 0, 0, .1)',
    level2: '0 0 2px rgba(0, 0, 0, .3)',
    level3: '0 0 3px rgba(0, 0, 0, .3)',
    level4: '0 0 5px rgba(0, 0, 0, .3)',
    level5: '0 0 8px rgba(0, 0, 0, .3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Ubuntu, sans-serif',
    text: 'Lato, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
  },
};

export default theme;
