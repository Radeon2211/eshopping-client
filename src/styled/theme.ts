const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#777',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#06AD96',
    greenDark: '#009680',
    red: '#F22613',
    redLight: '#F94D2A',
    redDark: '#985A5A',
  },
  shadows: {
    level1: '0 0 3px rgba(0, 0, 0, 0.03)',
    level2: '0 0 3px rgba(0, 0, 0, 0.3)',
    level3: '0 0 5px rgba(0, 0, 0, 0.3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Noto Sans, sans-serif',
    text: 'Nunito, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
    level5: 100000,
  },
  fontSizes: {
    level1: '1.3rem',
    level2: '1.4rem',
    level3: '1.5rem',
    level4: '1.7rem',
    level5: '2.2rem',
    level6: '2.6rem',
  },
  durations: {
    level1: 0.1,
    level2: 0.2,
    level3: 0.3,
  },
  lineHeights: {
    level1: '1',
    level2: '1.1',
    level3: '1.2',
    level4: '1.3',
    level5: '1.4',
    level6: '1.5',
  },
};

export type ThemeSpacing = keyof typeof theme.spacings;
export type ThemeZIndex = keyof typeof theme.zIndexes;
export type ThemeFontSize = keyof typeof theme.fontSizes;
export type ThemeDuration = keyof typeof theme.durations;
export type ThemeLineHeight = keyof typeof theme.lineHeights;

export default theme;
