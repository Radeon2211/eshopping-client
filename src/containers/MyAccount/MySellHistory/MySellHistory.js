import { useEffect } from 'react';
import { useSelector } from 'react-redux';
import * as SC from '../MyAccount.sc';
import { orderTypes } from '../../../shared/constants';
import Orders from '../../../components/Orders/Orders';
import Heading from '../../../components/UI/Heading/Heading';
import { scrollToTop } from '../../../shared/utility/utility';
import MetaDescriptor from '../../../components/MetaDescriptor/MetaDescriptor';
import useLastLocation from '../../../shared/useLastLocation';

export default function MySellHistory() {
  const lastLocation = useLastLocation();

  const sellHistory = useSelector((state) => state.auth.sellHistory);

  useEffect(() => {
    if (!lastLocation?.pathname.startsWith('/order/')) {
      scrollToTop();
    }
  }, [lastLocation]);

  return (
    <SC.OrdersWrapper>
      <MetaDescriptor
        title="Your sell history - E-Shopping"
        description="Check out your sell history"
      />
      <Heading $variant="h3">My sell history</Heading>
      <Orders orders={sellHistory} type={orderTypes.SELL_HISTORY} />
    </SC.OrdersWrapper>
  );
}
