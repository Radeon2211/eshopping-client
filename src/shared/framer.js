import theme from '../styled/theme';

export const stepFormVariants = {
  hidden: {
    opacity: 0,
  },
  visible: {
    opacity: 1,
    transition: { duration: theme.durations.level1, delay: theme.durations.level1 },
  },
};

export const backdropVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
  },
  visible: {
    opacity: 1,
    pointerEvents: 'initial',
    transition: { duration: theme.durations.level3 },
  },
};

export const modalVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
  },
  visible: {
    opacity: 1,
    transition: { delay: theme.durations.level2 },
    pointerEvents: 'initial',
  },
};

export const messageBoxVariants = {
  hidden: {
    pointerEvents: 'none',
    y: '-150%',
  },
  visible: {
    transition: { type: 'spring', duration: theme.durations.level2, stiffness: 100 },
    pointerEvents: 'initial',
    y: 0,
  },
};

export const dropdownVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
    y: '-10%',
  },
  visible: {
    opacity: 1,
    pointerEvents: 'initial',
    y: '0',
  },
};
