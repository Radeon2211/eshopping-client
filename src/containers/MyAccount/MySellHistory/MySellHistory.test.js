import React from 'react';
import { render, cleanup, waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { useLastLocation } from 'react-router-last-location';
import { ThemeProvider } from 'styled-components';
import thunk from 'redux-thunk';
import theme from '../../../styled/theme';
import MySellHistory from './MySellHistory';
import {
  createOrder,
  createTransactionAndOrderProdItem,
} from '../../../shared/testUtility/testUtility';
import { defaultScrollToConfig } from '../../../shared/constants';

const mockStore = configureMockStore([thunk]);

const products = [
  createTransactionAndOrderProdItem({
    productId: 'p1',
    sellerUsername: 'sellerUser',
    price: 80.2,
    quantity: 4,
    name: 'product1',
    photo: true,
  }),
];

const defaultStore = mockStore({
  auth: {
    sellHistory: [
      createOrder({
        products,
        id: 'o1',
        sellerUsername: 'sellerUser',
        buyerUsername: 'buyerUser',
        overallPrice: 320.8,
        createdAt: '2021-02-28T21:13:05.008Z',
      }),
    ],
    orderCount: 1,
  },
  ui: {
    isDataLoading: false,
  },
});

const defaultHistory = {
  listen: jest.fn(),
  createHref: jest.fn(),
  location: { pathname: '/my-account/sell-history', search: '?p=1' },
  replace: jest.fn(),
};

const setUp = () => {
  return render(
    <Provider store={defaultStore}>
      <Router history={defaultHistory}>
        <ThemeProvider theme={theme}>
          <MySellHistory />
        </ThemeProvider>
      </Router>
    </Provider>,
  );
};

jest.mock('react-router-last-location', () => ({
  useLastLocation: jest.fn(),
}));

afterEach(cleanup);

beforeAll(() => {
  window.scrollTo = jest.fn();
});

describe('<MySellHistory />', () => {
  describe('check how renders', () => {
    it('should render everything correctly with given default data', async () => {
      setUp();
      expect(screen.getAllByTestId('OrderList-single-order')).toHaveLength(1);
      await waitFor(() => {
        expect(document.title).toEqual('Your sell history - E-Shopping');
      });
    });
  });

  describe('check useEffect()', () => {
    it('should call scrollTo() if last rendered route was OrderDetails', () => {
      useLastLocation.mockImplementation(() => ({
        pathname: '/order/o1',
      }));
      setUp();
      expect(window.scrollTo).not.toHaveBeenCalled();
    });

    it('should NOT call scrollTo() if last rendered route was NOT OrderDetails', () => {
      useLastLocation.mockImplementation(() => ({
        pathname: '/cart',
      }));
      setUp();
      expect(window.scrollTo).toHaveBeenCalledWith(defaultScrollToConfig);
    });
  });
});
