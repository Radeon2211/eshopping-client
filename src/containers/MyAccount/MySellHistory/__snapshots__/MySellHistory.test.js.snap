// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MySellHistory /> check how renders should render everything correctly with given default data 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnbm eNBhAw"
  >
    <h3
      class="sc-dmlrTW zGXEE"
    >
      My sell history
    </h3>
    <section
      class="sc-hKgILt khuidQ"
    >
      <div
        class="sc-crrsfI jnkwPJ"
      >
        <div
          class="sc-hHftDr bPRUtw"
        >
          <label
            class="label"
            for="sortOption"
          >
            Sort
          </label>
          <div
            class="select css-2b097c-container"
            id="sortOption"
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-l640om-control"
            >
              <div
                class=" css-g1d714-ValueContainer"
              >
                <div
                  class=" css-1uccc91-singleValue"
                >
                  Date - from newest
                </div>
                <input
                  aria-autocomplete="list"
                  class="css-wmatm6-dummyInput-DummyInput"
                  id="react-select-2-input"
                  readonly=""
                  tabindex="0"
                  value=""
                />
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-ml9fjp-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-eCssSg kfCGWB"
          data-testid="InputPagination"
        >
          <a
            class="arrow hide-arrow"
            data-testid="InputPagination-left-arrow"
          >
            <div
              class="sc-jSgupP iHjayU"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
          <form
            class="form-number"
          >
            <input
              class="sc-iBPRYJ eTbKPo"
              data-testid="NumberInput-page"
              name="page"
              type="number"
              value="1"
            />
          </form>
          <span
            class="sc-gKsewC hpFlQD"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            of
          </span>
          <span
            class="sc-gKsewC hpFlQD"
            color=""
            data-testid="InputPagination-number-of-pages"
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            1
          </span>
          <a
            class="arrow hide-arrow"
            data-testid="InputPagination-right-arrow"
          >
            <div
              class="sc-jSgupP elniXq"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
        </div>
      </div>
      <div
        class="sc-jrAGrp fPUvfA"
      >
        <div
          class="sc-kEjbxe eqCUFO"
        >
          <div
            class="sc-dQppl jjWvYi"
            direction="row"
            spacing="3"
            width="100%"
            wrap="nowrap"
          >
            <span
              class="sc-gKsewC ISnTb"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              <span>
                buyer 
              </span>
              <a
                data-testid="OrderList-user-link"
              >
                <span
                  class="sc-iqHYGH fHWvCx"
                >
                  buyerUser
                </span>
              </a>
            </span>
            <span
              class="sc-gKsewC dAcVpy"
              color=""
              display="inline-block"
              overflow="visible"
              size="2"
              spacing="0px"
              transform=""
            >
              28 Feb 2021, 22:13
            </span>
          </div>
          <div
            class="sc-dQppl Ejhse sc-bqyKva eCKrkI"
            data-testid="TransactionAndOrderProdItem"
            direction="row"
            spacing="2"
            width="100%"
            wrap="nowrap"
          >
            <a
              data-testid="TransactionAndOrderProdItem-product-link-photo"
            >
              <div
                class="sc-dQppl bLxFyT sc-fodVxV gdFDCb"
                data-testid="ProductThumbnail"
                direction="row"
                height="5"
                spacing=""
                width="5"
                wrap="nowrap"
              >
                <img
                  alt="product1"
                  data-testid="ProductThumbnail-img"
                  src="http://192.168.1.109:4000/orders/o1/p1/photo"
                />
              </div>
            </a>
            <div
              class="sc-kstrdz cPEvH"
            >
              <span
                class="name"
              >
                <a
                  data-testid="TransactionAndOrderProdItem-product-link-name"
                >
                  product1
                </a>
              </span>
              <div
                class="sc-dQppl hYTqVV sc-hBEYos kYeIPx"
                direction="row"
                spacing=""
                width="100%"
                wrap="nowrap"
              >
                <span
                  class="sc-gKsewC vWWge"
                  color="#777"
                  data-testid="TransactionAndOrderProdItem-product-price-per-piece"
                  display="inline-block"
                  overflow="visible"
                  size="2"
                  spacing="0px"
                  transform=""
                >
                  4 x $80.20
                </span>
                <span
                  class="sc-gKsewC bSTHgV"
                  color=""
                  display="inline-block"
                  overflow="visible"
                  size="4"
                  spacing="0px"
                  transform=""
                >
                  $320.80
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-dQppl jleYWC"
            direction="row"
            spacing="3"
            width="100%"
            wrap="nowrap"
          >
            <a
              data-testid="OrderList-order-details-link"
            >
              <button
                class="sc-fFubgz cuQJWU"
                color="blue"
                type="button"
              >
                details
              </button>
            </a>
            <div>
              <span
                class="sc-gKsewC euBNFu"
                color="#777"
                display="inline-block"
                overflow="visible"
                size="3"
                spacing="1px"
                transform=""
              >
                TOTAL
              </span>
              <span
                class="sc-gKsewC bNImzD"
                color=""
                display="inline-block"
                overflow="visible"
                size="5"
                spacing="0px"
                transform=""
              >
                $320.80
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-pFZIQ fCNRZS"
      >
        <span
          class="sc-gKsewC rBmjh"
          color="#777"
          data-testid="PaginationCounter"
          display="inline-block"
          overflow="visible"
          size="1"
          spacing="0px"
          transform=""
        >
          1 - 1 of 1 order
        </span>
        <div
          class="sc-fubCfw hDVVIN"
          data-testid="NumberPagination"
        >
          <a
            class="arrow hide-arrow"
            data-testid="NumberPagination-left-arrow"
          >
            <div
              class="sc-jSgupP iHjayU"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
          <a
            class="number-link active"
            data-testid="NumberPagination-page1"
          >
            1
          </a>
          <span
            class="of"
          >
            of
          </span>
          <span
            class="of"
            data-testid="NumberPagination-number-of-pages"
          >
            1
          </span>
          <a
            class="arrow hide-arrow"
            data-testid="NumberPagination-right-arrow"
          >
            <div
              class="sc-jSgupP elniXq"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
